# Sensor data to spawn in Carla.
sensors: {
  gps: [
    {
      frame_id: "infrastructure/gps_link",
      transform: {x: -47.538, y: 20.806, z: 10.0, roll: 0.0, pitch: 0.0, yaw: 0.0},
      topic: "gps/fix",
      params:
      {
        noise_alt_bias: 0,   # default: 0
        noise_alt_stddev: 0, # default: 0
        noise_lat_bias: 0,   # default: 0
        noise_lat_stddev: 0, # default: 0
        noise_lon_bias: 0,   # default: 0
        noise_lon_stddev: 0, # default: 0
        noise_seed: 0,       # default: 0
        sensor_tick: 0       # default: 0
      }
    }
  ],
  rgb_camera: [
    {
     frame_id: "infrastructure/camera_0",
     transform: {x: -29.748, y: 34.803, z: 23.3, roll: 0.0, pitch: -42.592, yaw: -139.9},
     topic: "camera0",
     params:
     {
       bloom_intensity: 0.675,     # default: 0.675
       fov: 90.0,                  # default: 90.0
       fstop: 2.0,                 # default: 1.4
       image_size_x: 800,         # default: 800
       image_size_y: 600,          # default: 600
       iso: 100.0,                 # default: 100.0
       gamma: 2.2,                 # default: 2.2
       lens_flare_intensity: 0.1,  # default: 0.1
       sensor_tick: 0.0,           # default: 0.0
       shutter_speed: 200.0        # default: 200.0
     }
    },
    {
      frame_id: "infrastructure/camera_1",
      transform: {x: -62.718, y: 3.488, z: 36.503, roll: 0.0, pitch: -46.599, yaw: 49.184},
      topic: "camera1",
      params:
      {
        bloom_intensity: 0.675,     # default: 0.675
        fov: 90.0,                  # default: 90.0
        fstop: 1.8,                 # default: 1.4
        image_size_x: 800,          # default: 800
        image_size_y: 600,          # default: 600
        iso: 100.0,                 # default: 100.0
        gamma: 2.2,                 # default: 2.2
        lens_flare_intensity: 0.1,  # default: 0.1
        sensor_tick: 0.0,           # default: 0.0
        shutter_speed: 200.0        # default: 200.0
      }
    }
   ],
   lidar: [
     {
       frame_id: "velodyne_64",
       transform: {x: -47.538, y: 20.806, z: 10.0, roll: 0.0, pitch: 0.0, yaw: 0.0},
       topic: "velodyne_64_points",
       params:
       {
         channels: 64,                         # default: 32
         range: 120.0,                         # default: 10.0
         points_per_second: 1300000,           # default: 56000
         rotation_frequency: 60,               # default: 10.0
         upper_fov: 2.0,                       # default: 10.0
         lower_fov: -24.9,                     # default: -30.0
         horizontal_fov: 360.0,                # default: 360.0
         atmosphere_attenuation_rate: 0.004,   # default: 0.004 
         dropoff_general_rate: 0.45,           # default: 0.45
         dropoff_intensity_limit: 0.8,         # default: 0.8
         dropoff_zero_intensity: 0.4,          # default: 0.4 
         sensor_tick: 0.0,                     # default: 0.0
         noise_stddev: 0.01                    # default: 0.0
       }
     }
  ]
}

# semantic_camera: [
#   {
#     frame_id: "infrastructure/semantic_camera",
#     transform: {x: 23.388, y: -22.917, z: 20.056, roll: 0.0, pitch: -34.935, yaw: 136.943},
#     topic: "semantic_camera",
#     params:
#     {
#       fov: 90.0,                  # default: 90.0
#       sensor_tick: 0.0            # default: 0.0
#     }
#   }
# ]

# semantic_lidar: [
#   {
#     frame_id: "semantic_lidar_64",
#     transform: {x: -0.9, y: -0.566, z: 20.0, roll: 0.0, pitch: 0.0, yaw: 0.0},
#     topic: "semantic_lidar",
#     params:
#     {
#       channels: 64,              # default: 32
#       range: 100.0,              # default: 10.0
#       points_per_second: 250000, # default: 56000
#       rotation_frequency: 20,    # default: 10.0
#       upper_fov: 2.0,            # default: 10.0
#       lower_fov: -26.8,          # default: -30.0
#       horizontal_fov: 360.0,     # default: 360.0
#       sensor_tick: 0.0           # default: 0.0
#     }
#   }
# ]